helpers/helpers.go

```diff
@@ -13,80 +10,41 @@ import (
 
 // GetMapSlot returns the storage key slot for a holder.
 // Position is the index slot (storage index of amount balances map).
-func GetMapSlot(holder string, position int) ([32]byte, error) {
-	var slot [32]byte
-	hl, err := hex.DecodeString(TrimHex(holder))
-	if err != nil {
-		return slot, err
-	}
-	hl = common.LeftPadBytes(hl, 32)
-	posHex := fmt.Sprintf("%x", position)
-	if len(posHex)%2 == 1 {
-		posHex = "0" + posHex
-	}
-	p, err := hex.DecodeString(posHex)
-	if err != nil {
-		return slot, err
-	}
-	p = common.LeftPadBytes(p, 32)
-
-	hash := crypto.Keccak256(hl, p)
-	copy(slot[:], hash[:32])
-	return slot, err
+func GetMapSlot(holder common.Address, position int) [32]byte {
+	return crypto.Keccak256Hash(
+		common.LeftPadBytes(holder[:], 32),
+		common.LeftPadBytes(big.NewInt(int64(position)).Bytes(), 32),
+	)
 }
 
-// ValueToBalance takes a RLP encoded hexadecimal string and the number of decimals and returns
-// the balance as a big.Float number.
-func ValueToBalance(value []byte, decimals int) (*big.Float, *big.Int, error) {
+// ValueToBalance takes a big endian encoded value and the number of decimals
+// and returns the balance as a big.Rat (considering decimals) and big.Int
+// (not considering decimals).
+func ValueToBalance(value []byte, decimals int) (*big.Rat, *big.Int) {
```

Comment:
Do you see an easy way (one step) to transform a Rat number to big.Int absolute number (ignoring decimals but without knowing th enumber of decimals?). Such as Rat: "2.345" => Int: "2345". If that is possible then we can just return a single Rat value (no need for the Int).

Reply:
I think this can't be done, because rationals internally will reduce the quotient and reminder to the smallest equivalent.  For example, if you create a rational of 222/10 (which would be 22.2) internally it would be stored like 111/5.  Or 200/10 would become 20/1.  So the information about the number of decimals (which is required to get the integer back) is lost.